dashboard/app/models/stage.rb

```diff
@@ -44,4 +44,32 @@ def lesson_plan_pdf_url
   def lesson_plan_base_url
     CDO.code_org_url "/curriculum/#{script.name}/#{position}"
   end
+
+  def summarize
+    stage_data = {
+        script_id: script.id,
+        script_name: script.name,
+        script_stages: script.stages.to_a.count,
+        id: id,
+        position: position,
+        name: localized_name,
+        title: localized_title
+    }
+
+    if script.has_lesson_plan?
+      stage_data[:lesson_plan_html_url] = lesson_plan_html_url
+      stage_data[:lesson_plan_pdf_url] = lesson_plan_pdf_url
+    end
+
+    if script.hoc?
+      stage_data[:finishLink] = script.hoc_finish_url
+      stage_data[:finishText] = I18n.t('nav.header.finished_hoc')
+    end
+
+    levels = script.script_levels.to_a.select{|sl| sl.stage_id == id}
```

Comment:
If I remember this correctly (it's been a while since I wrote that) it was just code that I pulled in from elsewhere.  That means that either:
a) I never checked to see that stage.script_levels was correct in all cases, or
b) I found that it wasn't -- for instance maybe with scripts that had game/chapter definitions vs stage/level etc.
I don't think I did it for pure caching reasons.